### DISCUSSION TITLE
TeamPass Upgrade issue v3.0.9 to 3.1.0 & also v3.0.9 to v3.0.10

### DISCUSSION THREAD

**Discussion by jman1773** [nilsteampassnet/TeamPass#4011] [General]
Currently getting stuck upgrading at "Database Updates" section on my Ubuntu Server.

It gets to around Loop 70,000+ after about 12-24 hours & then my browser crashes and says "out of Memory" or crashed the browser this is on a Windows PC with 32GB RAM when running https://x.x.x.x/install/upgrade.

`Performing operation populateItemsTable_CreatedAt - Loop #70000 `

It is a big database with over 1,500+ passwords & for example the SQL dump file is about 40MB in size so I do have a lot of entries/items which most likely causing the issue.

Scrolling back down the full list before the crash/out of memory error everything has a green tick.

![image](https://github.com/nilsteampassnet/TeamPass/assets/156727185/680d0369-6cee-4bea-9068-6e7f05e77e5a)

After it failed on v3.0.9 to v3.0.10 I reverted everything (Database/files) back & then tried again on v3.0.9 to v3.1.0 but got the same issue just in case there was a bug upgrading from v3.0.9 to v3.0.10.

Two questions.

Q1) If the browser does crash does the action keep running in the background & if so is there any way to monitor the progress + continue/resume where I was up to before the browser crash?

Q2) If "No" to Q1. please advise how can I resolve this issue? (Eg. Maybe run Database update via mysql rather than the browser)

In case I need to run a mysql query/script here is my database version information "Database Version: 10.6.12-MariaDB-0ubuntu-.22.04.1 Ubuntu 22.04".

Many Thanks in advance :-)

**Comment by orochimarujr** [comment]
same problem, do you find any new solutions?

**Reply by jman1773** [reply]
@orochimarujr 
The only solutions was to switch to another Open Source solution which has better support such as VaultWarden (dani-garcia/vaultwarden) which is OpenSource Self-hosted solution it even supports bitwarden clients.

Vault Warden is a bit different in a few ways

- You will need to create Organisations & add users then put passwords inside it & folders don't work inside Organisations so you have to use collections I use like folders but you can restrict based on collections too so very useful 
- There is a built-in 2FA code generator so you don't need to use google Authenticator etc.
- You will also see there is no way to manually reset user passwords via admin portal without setting emergency Access for each user account.

The only other solution free self-hosted solution is passbolt Community Edition.
